### ISSUE TITLE
Improve the UI

### ISSUE THREAD

**Issue by rilwis** [proteusthemes/one-click-demo-import#54]
Hi,

Thanks for the great plugin. I'm planning to use it in my themes. But I think the current UI doesn't look good. I think we can improve it to make the list of demos look like the *Appearance > Themes* page, e.g. in a grid, so users have a quick look at each demo. Below each thumbnail can have demo name + preview button + import button.

This is an example from Total theme from WPExplorer which has a similar idea (just no preview and import button): http://wpexplorer-themes.com/total/docs/importing-sample-data/

**Comment by pagelab** [comment]
This was also referenced here: https://github.com/proteusthemes/one-click-demo-import/issues/51

**Comment by capuderg** [comment]
Hi,

yes, @pagelab, that's correct, I'll close this issue and leave the #51 as the main one.

@rilwis We will implement this layout in the next big plugin version (probably 2.0), where there will also be some other improvements, so thanks for letting us know that you would be interested in this kind of layout.

I hope I'll have time to implement everything in the next couple of weeks.

Take care!

**Comment by capuderg** [comment]
Hi,

we are close to the 2.0.0 release and in this release we also changed the multi predefined demo import layout. We added the grid layout instead of the dropdown select option.

I would like to ask you, if you could test it out and let me know, if you find any bugs or anything else... Here is the release candidate for the 2.0.0 version: https://dl.dropboxusercontent.com/u/14641788/OCDI/one-click-demo-import-2.0.0-RC.zip

We also added support for Redux framework and you can also group some demo imports into categories. So the filer for predefined demo imports would look something like this (categories and redux framework options were added):

`
function ocdi_import_files() {
	return array(
		array(
			'import_file_name'           => 'Demo Import 2',
                        'categories'                    => array( 'Category 1', 'Category 2' ),
			'import_file_url'            => 'http://www.your_domain.com/ocdi/demo-content2.xml',
			'import_widget_file_url'     => 'http://www.your_domain.com/ocdi/widgets2.json',
			'import_customizer_file_url' => 'http://www.your_domain.com/ocdi/customizer2.dat',
			'import_redux'               => array(
				array(
					'file_url'    => 'http://www.your_domain.com/ocdi/redux.json',
					'option_name' => 'redux_option_name',
				),
				array(
					'file_url'    => 'http://www.your_domain.com/ocdi/redux2.json',
					'option_name' => 'redux_option_name_2',
				),
			),
			'import_preview_image_url'   => 'http://www.your_domain.com/ocdi/preview_import_image2.jpg',
			'import_notice'              => __( 'A special note for this import.', 'your-textdomain' ),
		),
	);
}
add_filter( 'pt-ocdi/import_files', 'ocdi_import_files' );
`

And if you are using local import files (in the theme folder):

`
function ocdi_import_files() {
	return array(
		array(
			'import_file_name'             => 'Demo Import 2',
                        'categories'                    => array( 'Category 1', 'Category 2' ),
			'local_import_file'            => trailingslashit( get_template_directory() ) . 'ocdi/demo-content2.xml',
			'local_import_widget_file'     => trailingslashit( get_template_directory() ) . 'ocdi/widgets2.json',
			'local_import_customizer_file' => trailingslashit( get_template_directory() ) . 'ocdi/customizer2.dat',
			'local_import_redux'           => array(
				array(
					'file_path'   => trailingslashit( get_template_directory() ) . 'ocdi/redux.json',
					'option_name' => 'redux_option_name',
				),
				array(
					'file_path'   => trailingslashit( get_template_directory() ) . 'ocdi/redux2.json',
					'option_name' => 'redux_option_name_2',
				),
			),
			'import_preview_image_url'     => 'http://www.your_domain.com/ocdi/preview_import_image2.jpg',
			'import_notice'                => __( 'A special note for this import.', 'your-textdomain' ),
		),
	);
}
add_filter( 'pt-ocdi/import_files', 'ocdi_import_files' );
`

Let me know how it goes.

Take care!

**Comment by primozcigler** [comment]
@capuderg you are awesome! I want to encourage everyone to test `v2.0.0-RC` before it goes live!

**Comment by rilwis** [comment]
@capuderg This is an awesome news. I will test and let you know.

**Comment by rilwis** [comment]
I think the UI is much better! Thanks a lot for that.

By the way, I'd love to see the progress while importing, something similar to the log file (but simpler). It would help users a lot when the data is big (sometimes it takes 5 mins to import, without indicator of what's doing, users might think it's broken).

**Comment by capuderg** [comment]
Hi,

thanks! :)

the progress report will probably be the next bigger update after this one goes out. I can't say, when that will be, but it's on the to do list.

Could you maybe please create a new issue here, just so it's here and other can give opinions on...

Thanks and take care!

**Comment by ingeniousweb** [comment]
This is awesome,

Thank you so much for allowing this as requested, i have just did a quick test and the ui is really native and awesome, all works fine. However we are still missing the "Preview" button. I request you to please allow a preview url to be passed as well with so the users can preview the demo live (external link).

Thanks,
Munir

![screenshot-testsite dev 2016-12-02 15-46-28](https://cloud.githubusercontent.com/assets/5817882/20832405/f4579842-b8ab-11e6-8dbc-7a299d5c88a3.jpg)

**Comment by capuderg** [comment]
Hi Munir,

aha, I see. Could you please open a seperate issue for this feature. I will add this in the next plugin update (so this will not be available in the 2.0.0, but I'll add it for the 2.1.0).

Thanks and take care!

**Comment by ingeniousweb** [comment]
Hey @capuderg 

Done.  It's created here:
https://github.com/proteusthemes/one-click-demo-import/issues/66

Btw what are ETA for 2.0.0 and than what will be for 2.1.0 as i am looking to use this plugin in one of my coming theme/plugin.

Thanks,
Munir

**Comment by capuderg** [comment]
Thanks!

I will release the 2.0.0 today or tomorrow, depending, if no new bugs are found...

I have to do a bit more testing. And I would also like to ask you (all of you), if you could also import the content (test it out on your themes). Just to be sure, that everything works ok (so, not just the new grid UI, but the actual import as well).

**Comment by capuderg** [comment]
Hi,

just a quick update. I found two bugs while testing, so the plugin will not be updated this weekend.

I've pushed back the update to next weekend... I hope I'll have time to work on it and fix the issues. The main problem is that the customizer import is not working ok on WP version 4.7-RC2. Since the WP 4.7 release is just around the corner I don't want to update the plugin, if it is not working 100% with WP 4.7.

Here is the latest RC for our OCDI plugin, if someone wants to do some tests: https://dl.dropboxusercontent.com/u/14641788/OCDI/OCDI-2.0.0-RC2.zip

Take care!
